app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkDeselectNavigationListener.kt

```diff
@@ -32,7 +32,11 @@ class BookmarkDeselectNavigationListener(
      * Deselects all items when the user navigates to a different fragment or a different folder.
      */
     override fun onDestinationChanged(controller: NavController, destination: NavDestination, arguments: Bundle?) {
-        if (destination.id != R.id.bookmarkFragment || differentFromSelectedFolder(arguments)) {
+        val destinationIsNotBookmarkFragment =
```

Comment:
I know you were having some issues with the nav library but could you explain why this works and checking the destination ID does not?

Reply:
I wrote a brief explanation in the commit message (copied here).  I don't honestly know why the previous version didn't work.

> `resources.getResourceName(destination.id)` returned `org.mozilla.fenix.debug:id/bookmarkFragment`, which seems like it should have matched `R.id.bookmarkFragment`, but this was always false. This led to `onAllBookmarksDeselected` events being sent whenver bookmarks was opened.

Reply:
Oh oops sorry didn't see the commit message. Thanks!

Reply:
@ekager (recap from our discussion earlier).  This commit didn't do what I thought it did (thanks again for catching!).  I dropped the commit with those changes.  

I couldn't find a way to stop this event from being fired when first opening bookmarks, but I added a comment that at least mentions the behavior.